microcosm_postgres/encryption/encryptor.py

```diff
@@ -18,8 +18,16 @@ class SingleTenantEncryptor:
     A single tenant encryptor.
 
     """
-    def __init__(self, materials_manager: CryptoMaterialsManager):
-        self.materials_manager = materials_manager
+    def __init__(
+        self,
+        encrypting_materials_manager: CryptoMaterialsManager,
+        decrypting_materials_manager: CryptoMaterialsManager,
+    ):
+        self.encrypting_materials_manager = encrypting_materials_manager
+        self.decrypting_materials_manager = decrypting_materials_manager
```

Comment:
Note that this is done because the materials manager doesn't allow for separate key_providers for encryption vs. decryption. If we do decide to go fully to strict mode throughout, these could be combined again.